createAuthMiddleware issue. It is not exported in node_modules/better-auth/dist/plugins/index.mjs (?)

### Is this suited for github?

- [ ] Yes, this is suited for github

### To Reproduce

- I am running Astro v5 (probably does not matter, see below)
- In better-auth v1.5.3 this the error message running `astro run dev`
```ts
17:00:49 [ERROR] [MiddlewareCantBeLoaded] An unknown error was thrown while loading your middleware.
  Error reference:
    https://docs.astro.build/en/reference/errors/middleware-cant-be-loaded/
  Stack trace:
    at /root/repositories/productlionstorefront/node_modules/astro/dist/core/middleware/loadMiddleware.js:8:28
    [...] See full stack trace in the browser, or rerun with --verbose.
  Caused by:
  Export named 'createAuthMiddleware' not found in module '/root/repositories/productlionstorefront/node_modules/better-auth/dist/plugins/index.mjs'.
    at <parse> (:0)
    at linkAndEvaluateModule (native:1:11)
    at processTicksAndRejections (native:7:39)
```

- I am trying to use a hook in `auth.ts`.
- Documentation says the same.
- Example:
```ts
    hooks: {
        // https://www.better-auth.com/docs/reference/options#hooks
        after: createAuthMiddleware(async (ctx) => {
            //console.log(ctx);
        }),
    },
```

- Now in better-auth v1.4.21 the `createAuthMiddleware` is exported properly and it works fine:
```ts
import { createAuthEndpoint, createAuthMiddleware, optionsMiddleware } from "@better-auth/core/api";
```

- In better-auth v1.5.3 `createAuthMiddleware` is not exported. 
- If I manually change the `node_modules/better-auth/dist/plugins/index.mjs` by adding this into `index.mjs`:
```ts
import { createAuthEndpoint, createAuthMiddleware, optionsMiddleware } from "@better-auth/core/api";

//
// and I export it as (v1.4.21 has it) like: 
//

export { MULTI_SESSION_ERROR_CODES as ERROR_CODES, HIDE_METADATA, TWO_FACTOR_ERROR_CODES, USERNAME_ERROR_CODES, createAuthEndpoint, createAuthMiddleware, optionsMiddleware, admin, anonymous, auth0, bearer, captcha, createAccessControl, createJwk, customSession, deviceAuthorization, deviceAuthorizationOptionsSchema, emailOTP, generateExportedKeyPair, genericOAuth, getClient, getJwtToken, getMCPProtectedResourceMetadata, getMCPProviderMetadata, getMetadata, getOrgAdapter, gumroad, hasPermission, haveIBeenPwned, hubspot, jwt, keycloak, lastLoginMethod, line, magicLink, mcp, microsoftEntraId, multiSession, oAuthDiscoveryMetadata, oAuthProtectedResourceMetadata, oAuthProxy, oidcProvider, okta, oneTap, oneTimeToken, openAPI, organization, parseRoles, patreon, phoneNumber, role, signJWT, siwe, slack, testUtils, toExpJWT, twoFactor, twoFactorClient, username, verifyJWT, withMcpAuth };
```
- It works just fine.

- So I read the 1.5.0 tag release notes. There were some changes. But I do not see any difference in importing/using `createAuthMiddleware`. 
- What am I missing?

Thanks in advance.


### Current vs. Expected behavior

Current: `createAuthMiddleware` is unusable on v1.5.3.
Expected: Having `createAuthMiddleware` exported and usable.

### What version of Better Auth are you using?

1.5.3

### System info

```bash
{
  "system": {
    "platform": "linux",
    "arch": "x64",
    "version": "#12-Ubuntu SMP PREEMPT_DYNAMIC Fri Jan  9 20:46:52 UTC 2026",
    "release": "6.17.0-12-generic",
    "cpuCount": 16,
    "cpuModel": "Intel(R) Xeon(R) Silver 4112 CPU @ 2.60GHz",
    "totalMemory": "61.14 GB",
    "freeMemory": "9.08 GB"
  },
  "node": {
    "version": "v24.3.0",
    "env": "development"
  },
  "packageManager": {
    "name": "bun",
    "version": "1.3.10"
  },
  "frameworks": [
    {
      "name": "react",
      "version": "^19.2.4"
    },
    {
      "name": "astro",
      "version": "^5.18.0"
    }
  ],
  "databases": [
    {
      "name": "@libsql/client",
      "version": "^0.17.0"
    },
    {
      "name": "drizzle",
      "version": "^0.45.1"
    }
  ],
  "betterAuth": {
    "version": "Unknown",
    "config": null,
    "error": "process.env.DB_CONFIG_URL is undefined"
  }
}
```

### Which area(s) are affected? (Select all that apply)

Client, Backend, Types, Documentation, Package

### Auth config (if applicable)

```typescript
import { betterAuth } from "better-auth"
export const auth = betterAuth({
  emailAndPassword: {  
    enabled: true
  },
});
```

### Additional context

_No response_

--- a/packages/better-auth/package.json
+++ b/packages/better-auth/package.json
@@ -489,8 +489,11 @@
   },
   "dependencies": {
     "@better-auth/core": "workspace:*",
+    "@better-auth/drizzle-adapter": "workspace:*",
     "@better-auth/kysely-adapter": "workspace:*",
     "@better-auth/memory-adapter": "workspace:*",
+    "@better-auth/mongo-adapter": "workspace:*",
+    "@better-auth/prisma-adapter": "workspace:*",
     "@better-auth/telemetry": "workspace:*",
     "@better-auth/utils": "catalog:",
     "@better-fetch/fetch": "catalog:",
@@ -504,9 +507,6 @@
     "zod": "^4.3.6"
   },
   "devDependencies": {
-    "@better-auth/drizzle-adapter": "workspace:*",
-    "@better-auth/mongo-adapter": "workspace:*",
-    "@better-auth/prisma-adapter": "workspace:*",
     "@lynx-js/react": "^0.116.3",
     "@sveltejs/kit": "^2.53.3",
     "@tanstack/react-start": "^1.163.2",
@@ -538,9 +538,6 @@
     "vue": "^3.5.29"
   },
   "peerDependencies": {
-    "@better-auth/drizzle-adapter": "workspace:*",
-    "@better-auth/mongo-adapter": "workspace:*",
-    "@better-auth/prisma-adapter": "workspace:*",
     "@lynx-js/react": "*",
     "@prisma/client": "^5.0.0 || ^6.0.0 || ^7.0.0",
     "@sveltejs/kit": "^2.0.0",
@@ -562,15 +559,6 @@
     "vue": "^3.0.0"
   },
   "peerDependenciesMeta": {
-    "@better-auth/drizzle-adapter": {
-      "optional": true
-    },
-    "@better-auth/mongo-adapter": {
-      "optional": true
-    },
-    "@better-auth/prisma-adapter": {
-      "optional": true
-    },
     "@lynx-js/react": {
       "optional": true
     },

--- a/packages/prisma-adapter/package.json
+++ b/packages/prisma-adapter/package.json
@@ -41,9 +41,6 @@
       "default": "./dist/index.mjs"
     }
   },
-  "dependencies": {
-    "@prisma/client": "^7.4.1"
-  },
   "peerDependencies": {
     "@better-auth/core": "workspace:*",
     "@better-auth/utils": "^0.3.0",

